### PR TITLE
add pagination for repository and response message

### PR DESCRIPTION
**Pull request by RikoLearn**
ITNOA

related to #11 

Please review this branch:
**It contains pagination on repository and responses**

### CODE REVIEW

File: Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace BSN.Commons.Converters
+{
+    public class JsonForceDefaultConverter<T> : JsonConverter<T>
```
Comment (soroshsabz) [inline Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs:12]: Please add comments for this class

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿namespace BSN.Commons.Responses
+{
+    public class GenericResponseBase<T> : Response where T : class
```
Comment (soroshsabz) [inline Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs:7]: Please add comments for this class

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace BSN.Commons.Responses
+{
+    public class GenericResponseBaseWithPagination<T> : GenericResponseBase<T> where T : class
```
Comment (soroshsabz) [inline Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs:10]: Please add comments for this class

**Review by soroshsabz — CHANGES_REQUESTED** [review]
ITNOA

As you can see in https://www.codeguru.com/csharp/commenting-best-practices-in-c/#:~:text=Best%20Practices%20for%20C%23%20Commenting%20Below%20are%20some,precisely%20describe%20what%20a%20portion%20of%20code%20does.

- Only use comments for code snippets that are hard to understand or need clarification.
- Keep comments simple: remember their purpose. They should precisely describe what a portion of code does.
- Don’t be rude or mean in your comments. Be professional.
- If you change a portion of code, be certain to change the comments to!
- If you have multiline comments or multiple inline comments, make sure they are aligned for easier readability.
- Keep comment lines short. Use multiple lines if necessary versus cramming them all on one line.
- Use the same style of comments as the rest of your development team or that you use in your entire application.
- Avoid W.E.T. comments or comments that waste everyone’s time. Don’t explain something that is obvious.
- Avoid smelly comments. Smelly comments are comments that attempt to justify or explain bad code. Rewrite your code instead. One final note about C# comments – they are not only used to tell yourself – or other developers – what your code was intended to do.


And please see https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/ and https://swimm.io/learn/code-collaboration/comments-in-code-best-practices-and-mistakes-to-avoid/ . I think these links is very useful

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs (hunk 3 above)
Comment (RikoLearn) [reply Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs:10]: I resolved it

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs (hunk 2 above)
Comment (RikoLearn) [reply Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs:7]: I resolved it

File: Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs (hunk 1 above)
Comment (RikoLearn) [reply Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs:12]: I resolved it

Comment (soroshsabz) [reply Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs:12]: What is meaning of default form?

Please describe them in comments

thanks :)

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs (hunk 2 above)
Comment (soroshsabz) [reply Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs:7]: Please describe the main responsibility of this class

The cause of existence of this class must be describe carefully

thanks

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs (hunk 3 above)
Comment (soroshsabz) [reply Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs:10]: Please describe the main responsibility of this class

The cause of existence of this class must be describe carefully

thanks

**Review by soroshsabz — CHANGES_REQUESTED** [review]

File: Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs (hunk 1 above)
Comment (soroshsabz) [reply Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs:12]: The user must be know from your comments, that when he must use this class.

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs (hunk 3 above)
Comment (soroshsabz) [reply Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs:10]: The user must be know from your comments, that when he must use this class.

File: Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs (hunk 1 above)
Comment (RikoLearn) [reply Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs:12]: I think ,I resolved it

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs (hunk 3 above)
Comment (RikoLearn) [reply Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs:10]: I think ,I resolved it

File: Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs (hunk 1 above)
Comment (soroshsabz) [reply Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs:12]: Please add example for this

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs (hunk 3 above)
Comment (soroshsabz) [reply Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs:10]: Good ;)

File: Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs (hunk 1 above)
Comment (RikoLearn) [reply Source/BSN.Commons.PresentationInfrastructure/Converters/JsonForceDefaultConverter.cs:12]: I hope ,this was resolved😢

**Review by soroshsabz — APPROVED** [review]

**Pull request merged**